### ISSUE TITLE
Introducing code snippets

### ISSUE THREAD

**Issue by hdashnow** [elegant-scipy/elegant-scipy#140]
Some chapters start with a comment on where the code snippet came from, some don't. Some have only code from us.
- [x] Make sure all code additions are acknowledged in a uniform way at the start of each chapter (I think it should mention which bit they contributed as well)
- [x] We make a fuss in the preface about community code, but not all chapters have it, so maybe rephrase?

**Comment by hdashnow** [comment]
So, I'm trying to properly and uniformly attribute the code snippets. This is what I have so far. Is this correct? Am I missing any attributions? Maybe to smaller chunks later in chapters?

### Ch1 Elegant NumPy: The Foundation of Scientific Python

No attribution, code written by us.

### Ch2 Quantile normalization with NumPy and SciPy

No attribution, code written by us.

### Ch3 Networks of Image Regions with ndimage

**Code by Vighnesh Birodkar.**  
**Nominated by Juan Nunez-Iglesias.**

### Ch4 Frequency and the Fast Fourier Transform

No attribution, code written by us.

### Ch5 Contingency tables using sparse coordinate matrices

**Code by Juan Nunez-Iglesias,**  
**with suggestions by Jaime Frio and Warren Weckesser.**  
**Nominated by Andreas Mueller.**

### Ch6 Linear algebra in SciPy

No attribution, code written by us.

### Ch8 Big Data in Little Laptop with Toolz

**Code by Matt Rocklin.**  
**Nominated by Juan Nunez-Iglesias.**

**Comment by hdashnow** [comment]
@jni @stefanv can I get confirmation that this is all correct? If so I'll close this issue.

**Comment by jni** [comment]
@hdashnow yes, it is correct!

**Comment by jni** [comment]
thank you! =)
